packages/core-js/utils/ast/finders.js

```diff
@@ -12,11 +12,18 @@ function findContractDefinitions(astNode) {
 /**
  * Get the given contract by id on the given AST
  * @param {number} contractId
- * @param {import("solidity-ast").SourceUnit} astNode
+ * @param {import("solidity-ast").SourceUnit|import("solidity-ast").SourceUnit[]} astNodes
  * @returns {import("solidity-ast").ContractDefinition}
  */
-function findContractNodeWithId(contractId, astNode) {
-  for (const contractDefiniton of findAll('ContractDefinition', astNode)) {
+function findContractNodeWithId(contractId, astNodes) {
+  if (Array.isArray(astNodes)) {
+    for (const astNode of astNodes) {
+      const contractDefiniton = findContractNodeWithId(contractId, astNode);
+      if (contractDefiniton) return contractDefiniton;
+    }
+  }
```

Comment:
Added the option to find across several astNodes, as it is a  common pattern on our code.